test/integration/sanity/multi-case-url.test.js

```diff
@@ -0,0 +1,226 @@
+var expect = require('chai').expect,
+    sinon = require('sinon');
+
+describe('request url', function() {
+    var testrun;
+
+    describe('with lowercase', function() {
+        before(function(done) {
+            this.run({
+                collection: {
+                    item: [{
+                        request: {
+                            url: 'http://postman-echo.com/post?name=postman',
+                            method: 'POST'
+                        }},
+                    {
+                        request: {
+                            url: 'https://postman-echo.com/post?name=postman',
+                            method: 'POST'
+                        }}]
+                }
+            }, function(err, result) {
+                testrun = result;
+                done(err);
+            });
+        });
+
+        it('should complete the run', function () {
+            expect(testrun).to.be.ok;
+            sinon.assert.calledOnce(testrun.start);
+            sinon.assert.calledOnce(testrun.done);
+            sinon.assert.calledTwice(testrun.request);
+            sinon.assert.calledTwice(testrun.response);
+            sinon.assert.calledWith(testrun.done.getCall(0), null);
+        });
+
+        it('should process the http request', function() {
+            sinon.assert.calledWith(testrun.request.getCall(0), null);
+            sinon.assert.calledWith(testrun.response.getCall(0), null);
+
+            expect(testrun.response.getCall(0).args[2]).to.have.property('code', 200);
+        });
+
+        it('should process the https request', function() {
+            sinon.assert.calledWith(testrun.request.getCall(1), null);
+            sinon.assert.calledWith(testrun.response.getCall(1), null);
+
+            expect(testrun.response.getCall(1).args[2]).to.have.property('code', 200);
+        });
+
+        it('should normalise the protocol and hostname for http', function() {
+            var response = testrun.response.getCall(0).args[2].stream.toString();
+
+            expect(JSON.parse(response)).to.have.property('url')
```

Comment:
This (and all similar assertions in tests below) tie runtime tests to the structure of Echo responses. Also, compare the entire URL instead.

Reply:
so you're suggesting grouping them together under one it block ?? also we do already compare the entire url

Reply:
As discussed offline.